src/screens/ProfileScreen/ProfileScreen.js

```diff
@@ -1,21 +1,242 @@
-import React, { Component } from 'react';
-import { View, Text, TouchableOpacity } from 'react-native';
-import styles from './styles';
+import React, { useState, useEffect } from 'react';
+import { View, Text, TextInput, TouchableOpacity, SafeAreaView, Image, ScrollView, Platform } from 'react-native';
 import { firebase } from '../../config/config';
+import ImagePicker from 'react-native-image-crop-picker';
+import styles from './styles';
+
+const ProfileScreen = () => {
+
+    const [name, setName] = useState("")
+    const [occupation, setOccupation] = useState("")
+    const [interest, setInterest] = useState("")
+    const [country, setCountry] = useState("")
+    const [editfield, setEditfield] = useState(false)
+    const [image, setImage] = useState(null)
+    const [displayImage, setDisplayImage] = useState("https://upload.wikimedia.org/wikipedia/commons/thumb/a/ac/No_image_available.svg/600px-No_image_available.svg.png");
+
+    const userID1 = firebase.auth().currentUser.uid
+    const ref = firebase.firestore().collection("users").doc(userID1);
+
+    useEffect(() => {
+        ref.get().then((doc) => {
+            if (doc.exists) {
+                console.log("Document data:", doc.data());
+                setName(doc.data().fullName);
+                setInterest(doc.data().interest)
+                setCountry(doc.data().country)
+                setOccupation(doc.data().occupation)
+                setDisplayImage(doc.data().downloadURL)
+                console.log(name);
+                
+            } else {
+                console.log("No such document!");
+            }
+        }).catch((error) => {
+            console.log("Error getting document:", error);
+        });     
+  }, [])
+
+    const uploadImage = async () => {
+      if(image == null){
+        savePostData(displayImage) 
+        return
+      }
+        const uri = image;
+        const childPath = `post/${firebase.auth().currentUser.uid}/${Math.random().toString(36)}`;
```

Comment:
Used Random string function in order to create a unique URL. I guess no inbuild function is present for generating a URL while uploading the blob to firebase storage, So I used this one. Just wanted to store the images with a unique URL, and found this method effective in javascript. Let me know if I am right.

Reply:
Gives me path likes this `post/ULlNNS8WhdXNHszcAt1rwHwTB1W2/0.d2ref89f2f5`

Reply:
what is the need of a unique URL here? It's the profile picture. Why can't we use post/uid/profile_picture.png here? Otherwise in every profile picture update, the user upload a image which is quite not correct and will increase the storage usage unnecessary. what we should do here is replace the existing profile picture with the new one.

Reply:
oh ok! got it!